src/pam_python.c

```diff
@@ -127,6 +127,14 @@ static char libpython_so[]	= LIBPYTHON_SO;
 static void initialise_python(void)
 {
 #if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 204
+#if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 312
+  PyConfig config;
+  PyConfig_InitPythonConfig(&config);
```

Comment:
```
PyConfig config;
PyConfig_InitIsolatedConfig(&config);   // note Isolated (!)!!!!!!
config.isolated = 1;
config.write_bytecode = 0;
config.use_environment = 0;
config.site_import = 1;
config.user_site_directory = 0;
config.install_signal_handlers = 0;
PyStatus status = Py_InitializeFromConfig(&config);
if (PyStatus_Exception(status)) {
   PyConfig_Clear(&config);
   if (!PyStatus_IsExit(status)) {
      Py_ExitStatusException(status);
   }
   abort(); // I don't know what to do here.
}
PyConfig_Clear(&config);
```

Reply:
Please revisit this MR. I have amended as suggested. Sorry for the long delay.

Reply:
@socketpair ^^^